### PR TITLE
Generate uses

### PR DESCRIPTION
**Pull request by EPashkin**
Current only common and parents

### CODE REVIEW

**Comment by gkoz** [comment]
Hmm, I don't see why the `Info` can't be borrowed mutably.

**Comment by EPashkin** [comment]
I think you can pass &mut to many subfunction, but did not checked.

**Comment by EPashkin** [comment]
Removed RefCell.
Don't merge now, i add interfaces in hour or two

**Comment by EPashkin** [comment]
Now only build errors in Gtk is `pango::Layout`, and 36 unresolved `ffi::*` :smile:

**Comment by EPashkin** [comment]
Ready to merge

File: src/analysis/functions.rs [hunk 1]
```diff
@@ -14,21 +15,27 @@ pub struct Info {
...
 
-fn analyze_function(env: &Env, type_: &library::Function, class_tid: library::TypeId) -> Info {
+fn analyze_function(env: &Env, type_: &library::Function, class_tid: library::TypeId,
+    used_types: &mut HashSet<String>) -> Info {
     let mut commented = false;
 
-    let ret = if type_.ret.typ == Default::default() { None } else { Some(type_.ret.clone()) };
+    let ret = if type_.ret.typ == Default::default() { None } else {
+        let _ = used_rust_type(&env.library, type_.ret.typ)
+            .map(|s| used_types.insert(s));
```
Comment (gkoz) [inline src/analysis/functions.rs]: `Result::ok` seems a good fit for this: `used_rust_type(&env.library, type_.ret.typ).ok().map(|s| used_types.insert(s));`

Comment (EPashkin) [inline src/analysis/functions.rs]: Yes, i fix that

**Comment by EPashkin** [comment]
Done.

**Comment by EPashkin** [comment]
Also 35 waiting

**Comment by EPashkin** [comment]
Don't understand why no "unused result" warning for `Option<()>`

**Comment by gkoz** [comment]
Only the `Result` causes the "unused" warning. Using `map` here seems a bit of abuse anyway because the value isn't actually mapped, so a good old `if` could be better. But that's not too important.

**Pull request merged**